### PR TITLE
Compatibility, deprecation and security policies

### PR DESCRIPTION
**Pull request by savchenko**
All-in-one. Pending @djmitche's GPG key.

Closes https://github.com/taskchampion/taskchampion/issues/210

### CODE REVIEW

File: POLICY.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+1. Major versions represent significant change and may be incompatible with previous major release.
+2. Minor versions are always backwards compatible and might add some new functionality.
+3. Patch versions should not introduce any new functionality and do what name implies &mdash; fix bugs.
+
+As there are no major releases yet, we do not support any older versions. Users are encouraged to use the latest release.
+
+## ABI policy
+
+1. We target stable `rustc`.
+2. TaskChampion will never upgrade any backend to a non-compatible version without explicit user's request.
```
Comment (djmitche) [inline POLICY.md:14]: what is a "backend"?

File: POLICY.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## API policy
```
Comment (djmitche) [inline POLICY.md:16]: What constitutes an API in this sense?  CLI?  Rust API?  Replica/server HTTP API?

File: POLICY.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+3. TaskChampion always uses UTF-8.
+
+## Exit codes
+
+- `0` No errors, normal exit.
+- `1` Generic error.
+- `2` Never used to avoid conflicts with Bash.
+- `3` Unable to execute with the given parameters.
+- `4` I/O error.
+- `5` Database error, irrespective of the backend used.
```
Comment (djmitche) [inline POLICY.md:39]: This looks good -- let's get an issue on file to support it.  I think this applies only to the CLI, so perhaps mention that here?

File: POLICY.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Command-line interface
+
+Considered to be part of the API policy.
```
Comment (djmitche) [inline POLICY.md:43]: Ah, ok.. then why's it down here? :)

File: POLICY.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+# Security
+
+To report a vulnerability, please contact [dustin@cs.uchicago.edu](dustin@cs.uchicago.edu), you may use GPG public-key `KEY-ID` which is available [here](#). Initial response is expected within ~48h.
```
Comment (djmitche) [inline POLICY.md:47]: https://keybase.io/djmitche

File: POLICY.md (hunk 1 above)
Comment (savchenko) [reply POLICY.md]: Re-written in light of https://github.com/taskchampion/taskchampion/issues/175

File: POLICY.md (hunk 2 above)
Comment (savchenko) [reply POLICY.md:16]: CLI and HTTP, Rust would be ABI I believe. Re-shuffled text to make it clearer.

File: POLICY.md (hunk 3 above)
Comment (savchenko) [reply POLICY.md]: https://github.com/taskchampion/taskchampion/issues/217 and mentioned in the https://github.com/savchenko/taskchampion/commit/db9d5483e83fe1d4c76842b29cb8418c39cc8932

File: POLICY.md (hunk 4 above)
Comment (savchenko) [reply POLICY.md]: Moved higher, please see the latest revision.

File: POLICY.md (hunk 5 above)
Comment (savchenko) [reply POLICY.md]: Added

File: docs/assets/cgi/LICENSE.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+Copyright (C) Andrew Savchenko - All Rights Reserved
+All files within this folder are proprietary and reserved for the use by TaskChampion project.
```
Comment (djmitche) [inline docs/assets/cgi/LICENSE.md:2]: It looks like these snuck into this PR, perhaps because it's based on `main`?

**Review by djmitche — APPROVED** [review]
Looks good!  I will merge this as-is, but would still like to find a way to land the source files in #218.

File: docs/assets/cgi/LICENSE.md (hunk 6 above)
Comment (savchenko) [reply docs/assets/cgi/LICENSE.md:2]: Indeed. This belongs to https://github.com/taskchampion/taskchampion/pull/218

**Pull request merged**